pylint/pyreverse/diagrams.py

```diff
@@ -214,20 +214,35 @@ def extract_relationships(self) -> None:
                     self.add_relationship(obj, impl_obj, "implements")
                 except KeyError:
                     continue
-            # associations link
-            for name, values in list(node.instance_attrs_type.items()) + list(
+            # associations & aggregations links
+
+            for name, values in list(node.aggregations_type.items()):
+                for value in values:
+                    self.assign_association_relationship(
+                        value, obj, name, "aggregation"
+                    )
+
+            for name, values in list(node.associations_type.items()) + list(
                 node.locals_type.items()
             ):
+
                 for value in values:
-                    if value is astroid.Uninferable:
-                        continue
-                    if isinstance(value, astroid.Instance):
-                        value = value._proxied
-                    try:
-                        associated_obj = self.object_from_node(value)
-                        self.add_relationship(associated_obj, obj, "association", name)
-                    except KeyError:
-                        continue
+                    self.assign_association_relationship(
+                        value, obj, name, "association"
+                    )
+
+    def assign_association_relationship(
+        self, value: Any, obj: ClassEntity, name: str, type_relationship: str
```

Comment:
Can we specify this? `object_from_node` only accepts `NodeNG`. Since `astroid` doesn't have a `py.typed` `mypy` doesn't understand this, but it would cause problems when it (finally) does.

Reply:
Added type hint for obj :+1: